### PR TITLE
feat(pubsub): re-implement the pubsub engine for standalone client

### PR DESCRIPTION
**Pull request by vlady-kotsev**
### Re-implement  standalone client Pubsub
<hr/>
This PR is part 1 of 3 Pubsub PRs.
<br/>
<br/>
Next: 
<br/>
 2. Re-implement the pubsub engine for cluster client
 <br/>
 3. Wire the pubsub engine 

---

> [!NOTE]
> **Medium Risk**
> Large new concurrent networking core (reconnect, confirmation races, shared fan-out) with broad test coverage, but it is not yet wired as the sole client path in this PR.
> 
> **Overview**
> Introduces a new **`internal/pubsub`** engine (part 1 of the standalone pub/sub rewrite): a **`Manager`** multiplexes every subscription on one shared connection, with per-subscriber **handles**, registries for channel/pattern/shard namespaces, and a **reply ledger** so subscribe confirmations, pongs, and error replies are attributed correctly (including targeted vs broadcast replay and cluster slot-migration heuristics).
> 
> The manager adds **reconnect with subscription replay**, optional **health-check PINGs**, **pending-subscribe resync**, **maintenance handoff** redirect, **slow-consumer drops**, and **SSUBSCRIBE** commands split per hash slot. Public-facing types live behind **`PubSuber`** / **`PubSubManagerer`**; delivery uses **`Events`**, **`Channel`**, and **`ChannelWithSubscriptions`** with runtime config via **`ChannelOption`**.
> 
> Coverage is heavy: pipe-based **fake Redis** unit tests, race/confirmation edge cases, and an external **`pubsub_test`** integration suite against a live Redis. Separately, **cluster full-duplex internal tests** now skip unless **`CLUSTER INFO`** reports **`cluster_state:ok`**, not only a successful **`PING`**.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 4c553d0cbaa1ba7197cc94fddcf4e5fa6ed95abb. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: internal/pubsub/pubsub_handle.go [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+
+// Channel returns a Go channel for concurrently receiving messages,
+// created on the first call and closed together with the handle.
+// Options run before h.mu is taken (h.mu and the manager's mu never
+// nest). It cannot be combined with ChannelWithSubscriptions.
+func (h *handle) Channel(opt ...ChannelOption) <-chan *Message {
+	for _, o := range opt {
+		o(h)
+	}
+	size, sendTimeout := h.m.consumerSettings()
```
[the remark this answers is not in the document]
Comment (vlady-kotsev) [reply internal/pubsub/pubsub_handle.go:60]: This is expected
